server/src/main/java/org/elasticsearch/search/builder/SearchQueryWrapperBuilder.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.builder;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.index.query.AbstractQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryRewriteContext;
+import org.elasticsearch.index.query.Rewriteable;
+import org.elasticsearch.index.query.SearchExecutionContext;
+import org.elasticsearch.search.query.SearchQueryWrapper;
+import org.elasticsearch.usage.SearchUsage;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * {@code SearchQueryBuilder} is a wrapper class for containing all
+ * the information required to perform a single search query
+ * as part of a series of multiple queries for features like ranking.
+ * It's expected to typically be used as part of a {@link java.util.List}.
+ */
+public class SearchQueryWrapperBuilder implements ToXContent, Writeable, Rewriteable<SearchQueryWrapperBuilder> {
+
+    public static SearchQueryWrapperBuilder parseXContent(XContentParser parser, SearchUsage searchUsage) throws IOException {
+        XContentParser.Token token;
+        String currentFieldName = null;
+
+        QueryBuilder queryBuilder = null;
+
+        if (parser.currentToken() == null) {
+            parser.nextToken();
+        }
+
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+            } else if (token == XContentParser.Token.START_OBJECT) {
+                if (SearchSourceBuilder.QUERY_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
```

Comment:
Should we fail add fail in other case (where QUERY_FIELD doesn't match), otherwise we accept arbitrary request like this:
```js
{
    "queries": [
        {
            "query": {
                "match": {
                    "k1": "blue"
                }
            }, 
            "rrf" : 2
        }
    ]
}
```

Reply:
Good catch! I'll update this to not accept extraneous parameters.

Reply:
I agree with Mayya, we should restrict this until we are sure more options are added.

Reply:
I've changed this to a ConstructingObjectParser which should create stronger consistency in both error messages and disallowing unknown fields.